ldms/src/core/ldms_xprt.c

```diff
@@ -350,6 +355,8 @@ static void send_dir_update(struct ldms_xprt *x,
 	size_t json_data_len;
 	struct ldms_reply *reply = NULL;
 
+	if (!ldms_xprt_connected(x))
```

Comment:
@tom95858 Why do we test whether the state is connected here? Is it to prevent passing NULL zap endpoint to zap_send() at line 397?

Reply:
The point is to avoid submitting a bunch of I/O on an endpoint that is closing. This is really a uGNI 'work-around'. Because we defer disconnect on the uGNI transport, the daemon will continue to submit requests to an endpoint that is effectively closed. This causes many PostRdma requests to be submitted on an endpoint (and associated CQ) that leak descriptors due to a bug in uGNI. So the idea is to detect this as early as possible and minimize the # of descriptors that get leaked.

It may well be that this logic is no longer required given that we now track (or try to) leaked descriptors and opportunistically re-create the CQ.

Reply:
Got it! Thanks for the explanation.